### PR TITLE
Fix origin offseting if i or j is not present in field for gtc backends

### PR DESCRIPTION
**Pull request by gronerl**
- `sid::shift_sid_origin` with "unnamed" indices needs to go before renaming
- fix use of `sid::get_strides` (`sid::sid_get_strides` is not part of the public API, note the extra `sid_`)

### CODE REVIEW

**Comment by gronerl** [comment]
bors try

**Comment by gronerl** [comment]
bors try

**Comment by havogt** [comment]
Before we merge this, we need to fully understand the problem.
- I understand the part about `sid_shift_origin`. @anstaf Let's discuss if we need another API for `sid_shift_origin` that takes also `stride`. It's not strictly necessary, the implementation in this PR is ok.
- I don't fully understand. Can we make a reproducible? It's either a bug, then your workaround will break once the bug is fixed, or it's wrong use of sid.

**Comment by gronerl** [comment]
Why do you think it will break once the (possible?) bug is fixed? When applying sid_shift_origin, the dimensions are always in the right order.

**Comment by gronerl** [comment]
Ah I guess your second bullet refers to the changes in the dace backend.

**Comment by havogt** [comment]
Sorry, yes, in the second point I forgot to link it to the second problem.
Here my updated assessment:
- The fix for the first part is correct, as discussed in the meeting. We could improve the code by using a hymap for the offsets.
- After looking closer, it seems there is a bug in the SID implementation, regarding your second problem (DaCe backend).

File: src/gt4py/backend/gtc_backend/dace/backend.py [hunk 1]
```diff
@@ -160,27 +160,24 @@ def generate_dace_args(self, gtir, sdfg):
                 symbols[f"__{name}_J_stride"] = str(array.shape[2])
                 symbols[f"__{name}_I_stride"] = str(array.shape[1] * array.shape[2])
             else:
-                data_ndim = len(array.shape) - sum(array_dimensions(array))
+                dims = [dim for dim, select in zip("IJK", array_dimensions(array)) if select]
+                data_ndim = len(array.shape) - len(dims)
 
                 # api field strides
                 fmt = "gt::sid::get_stride<{dim}>(gt::sid::sid_get_strides(__{name}_sid))"
+
```
Comment (havogt) [inline src/gt4py/backend/gtc_backend/dace/backend.py:168]: in the line above there is a bug: it should be `gt::sid::get_strides`. `sid_` is wrong.

**Review by havogt — CHANGES_REQUESTED** [review]

**Comment by gronerl** [comment]
bors try

**Review by havogt — APPROVED** [review]

**Pull request merged**